src/test/scala/mesosphere/marathon/api/akkahttp/v2/GroupsControllerTest.scala

```diff
@@ -62,6 +80,116 @@ class GroupsControllerTest extends UnitTest with ScalatestRouteTest with Inside
+      val f = new Fixture(groupManager = groupManager)
+
+      Get(Uri./.withPath(Path("/groupname/versions"))) ~> f.groupsController.route ~> check {
+        responseAs[String] should be ("[ \"2015-04-09T12:30:00.000Z\", \"2015-04-09T12:30:00.000Z\" ]")
+      }
+    }
+
+    "list all versions of root group" in {
+      val clock = new SettableClock()
+      val groupManager = mock[GroupManager]
+      val groupVersions = Seq(clock.now())
+      groupManager.versions(PathId.empty) returns Source(groupVersions)
+      groupManager.group(PathId.empty) returns Some(createGroup(PathId.empty))
+      val f = new Fixture(groupManager = groupManager)
+
+      Get(Uri./.withPath(Path("/versions"))) ~> f.groupsController.route ~> check {
+        responseAs[String] should be ("[ \"2015-04-09T12:30:00.000Z\" ]")
+      }
+    }
+  }
+
+  "Version detail" should {
+
+    "show app in a given version" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroupVersion(any, any, any, any) returns Future.successful(Some(GroupInfo(createGroup(PathId("/test/group")), None, None, None)))
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/test/group/versions/2017-10-30T16:08:53.852Z"))) ~> f.groupsController.route ~> check {
+        responseAs[String] should include (""""id" : "/test/group"""")
+      }
+    }
+
+    "show app in a given version for root group" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroupVersion(eq(PathId.empty), any, any, any) returns Future.successful(Some(GroupInfo(createGroup(PathId.empty), None, None, None)))
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/versions/2017-10-30T16:08:53.852Z"))) ~> f.groupsController.route ~> check {
+        responseAs[String] should include (""""id" : "/"""")
+      }
+    }
+
+    "reject for app and version that does not exist" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroupVersion(any, any, any, any) returns Future.successful(None)
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/groupname/versions/2017-10-30T16:08:53.852Z"))) ~> f.groupsController.route ~> check {
+        rejection should be (EntityNotFound.noGroup("groupname".toRootPath, Some(Timestamp("2017-10-30T16:08:53.852Z"))))
+      }
+    }
+  }
+
+  "Create a group" should {
+    "fail with rejection for group that already exists" in {
+      val groupManager = mock[GroupManager]
+      groupManager.rootGroup() returns createRootGroup()
+      val f = new Fixture(groupManager = groupManager)
+      val entity = HttpEntity("{}").withContentType(ContentTypes.`application/json`)
+
+      Post(Uri./.withPath(Path("/")), entity) ~> f.groupsController.route ~> check {
+        rejection should be (Rejections.ConflictingChange(Message(s"Group / is already created. Use PUT to change this group.")))
```

Comment:
Check the `LeaderControllerTest`. One can do

```scala
rejection shouldBe a[Rejections.ConflictingChange]
inside(rejection) {
  case Rejections.ConflictingChang(error) =>
    error.message should be("Group / is already created. Use PUT to change this group.")
}
```

This makes the error messages nicer.

Please not that scapegoat adds a warning for `s"..."` if there is not string interpolation.